app/main/views/frameworks.py

```diff
@@ -274,6 +275,80 @@ def framework_start_supplier_declaration(framework_slug):
 
 
 @main.route('/frameworks/<framework_slug>/declaration', methods=['GET'])
+@login_required
+def framework_supplier_declaration_overview(framework_slug):
+    framework = get_framework(data_api_client, framework_slug, allowed_statuses=['open'])
+
+    sf = data_api_client.get_supplier_framework_info(current_user.supplier_id, framework_slug)["frameworkInterest"]
+    # ensure our declaration is at least (probably) a dict
+    sf["declaration"] = sf.get("declaration") or {}
+
+    content = content_loader.get_manifest(framework_slug, 'declaration').filter(sf["declaration"])
+
+    # we must perform an actual validation for each section rather than rely on .answer_required as the latter won't
+    # take into account declarations custom question dependencies
+    sections_errors = OrderedDict(
```

Comment:
I much prefer declarative over imperative. With my one, I can immediately discard the possibility that the block has any side effects (unless it's written *very* strangely). In imperative style it's very hard to tell between many subtly different patterns which may or may not have side effects.